accepted/future-releases/nnbd/feature-specification.md

```diff
@@ -1008,6 +1011,33 @@ void main() {
 }
 ```
 
+### Constant assertions
+
+Currently assertions are enabled only when the compiler is passed an `enable-asserts` flag.
+With null safety, when an assert is evaluated as part of a constant invocation of a `const`
+constructor, any `assert`s in the initializer list are evaluated. 
+The `enable-asserts` flag then only applies to *run-time assertions*.
+
+More precisely, a assert _s_ of the form <code>assert(_e_<sub>1</sub>, _e_<sub>2</sub>)</code>
+is executed as follows:
+  * If the execution of _s_ happens during evaluation of a const invocation of the 
+    surrounding `const` constructor, then _e_<sub>1</sub> is evaluated to a constant value _c_.
```

Comment:
How about `then it is a compile-time error if _e_<sub>1</sub> is not a constant expression with the value `true``?

Then we don't have to mention 'completes normally' as if it's part of the constant evaluation semantics.

Reply:
Good point. We already do the "compile-time error if expression would throw" for constant evaluation.